massa-graph/src/block_graph.rs

```diff
@@ -2678,6 +2675,46 @@ impl BlockGraph {
             })
     }
 
+    /// get the latest blockclique (or final) block ID at a given slot, if any
+    pub fn get_latest_blockclique_block_at_slot(&self, slot: &Slot) -> BlockId {
+        let (mut latest_final_block_id, mut latest_final_block_period) = self
+            .latest_final_blocks_periods
+            .get(slot.thread as usize)
+            .unwrap_or_else(|| panic!("unexpected not found latest final block period"));
+
+        self.max_cliques
+            .iter()
+            .filter(|c| c.is_blockclique)
```

Comment:
Note that this will iterate on all cliques to filter them.
I recommend doing `let blockclique = &self.max_cliques.find(|c| c.is_blockclique).expect("no blockclique found").block_ids;` which is short-circuiting (will not scan the whole container but stop at the first found element)

Reply:
See https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find